### PR TITLE
feat(ios): Share a link for Messenger on iOS

### PR DESCRIPTION
**Pull request by ken0nek**
# Overview

Add a feature to share URLs for Messenger on iOS

I want to port this implementation into this repo.

```js

Linking.openURL('fb-messenger://share?link=https://github.com/react-native-share/react-native-share');
```

ref: https://tools.applemediaservices.com/app/454638411?country=us

# Test Plan

### CODE REVIEW

**Review by mikehardy — APPROVED** [review]
This looks good to me (for a rebase merge even, as the two commits are separate and clean)
Definitely compiles locally but I didn't test it because I don't have an app that can do it

Any thoughts on adding it to the list of examples @ken0nek ? https://github.com/react-native-share/react-native-share/blob/main/example/App.js

**Comment by ken0nek** [comment]
Thank you for the review!

I would like to add an example of this feature.

**Comment by MateusAndrade** [comment]
> Thank you for the review!
> 
> I would like to add an example of this feature.

Nice! Do you think we can add the example separadetly?

**Review by MateusAndrade — APPROVED** [review]

File: ios/RNShare.m [hunk 1]
```diff
@@ -84,6 +85,7 @@ - (NSDictionary *)constantsToExport
     @"INSTAGRAMSTORIES": @"instagramstories",
     @"TELEGRAM": @"telegram",
     @"EMAIL": @"email",
+    @"MESSENGER": @"messanger",
```
Comment (LeongXianJun) [inline ios/RNShare.m:88]: fyi, typo here causes failure to access messenger sharing if someone is using the enum exported in their shareSingle method

**Pull request merged**